### ISSUE TITLE
OpenStruct is deprecated

### ISSUE THREAD

**Issue by JamieMagee** [dependabot/dependabot-core#11163]
### Code improvement description

As of Ruby 3.4.0 OpenStruct (AKA `ostruct`) has been deprecated and is no longer loaded by default. In Ruby 3.5.0 it will be completely removed[^1].

There are currently a couple of places where we use OpenStruct:
 
- `common/lib/dependabot/git_metadata_fetcher.rb`
- `dependabot/file_fetchers/base.rb`
- `metadata_finders/base/changelog_finder.rb`
- `metadata_finders/base/release_finder.rb`

These should be migrated to, preferably, ruby classes or hashes.

If we don't want to do this, for some reason, we can also add an explicit reference to the `ostruct` gem[^2]

[^1]: https://bugs.ruby-lang.org/issues/20309
[^2]: https://rubygems.org/gems/ostruct

**Comment by daandj** [comment]
Hi, I have never contributed to an open source project before, but would like to start. Is this issue something I could work on and try and solve?

**Comment by JamieMagee** [comment]
@daandj yes, certainly. It should be fairly straightforward if you have experience working in Ruby, and we can help guide you through the open source contribution part.

**Comment by corsonknowles** [comment]
@JamieMagee @daandj I'm offering this solution:

* https://github.com/dependabot/dependabot-core/pull/14011
